pkg/frontend/openshiftversions_list.go

```diff
@@ -42,13 +41,5 @@ func (f *frontend) getEnabledInstallVersions(ctx context.Context) []*api.OpenShi
 	}
 	f.mu.RUnlock()
 
-	if len(versions) == 0 {
```

Comment:
This is an interesting symptom of this change - I feel like we should have this case outlined somewhere, right? Maybe we decide that it lands in the RP release pipeline or something? We need to guarantee that we'll always return at least one version (defaultversion) to the customer at all times and prevent rollouts that would make this possible (we could also do this at an RP level, for example, we exit `1` if no default version is found, and that happens way before a request would reach `frontend`).

Reply:
That brings up an important point: we need to run the **Update ARO OpenShift Versions** pipelines to deploy the default flag _before_ we release an RP with this change so that the scenario you're talking about can't happen.  If we're releasing the RP next week, then we should either start a version update rollout this week or hold this PR until September.

I did add a [check to "aro update-versions"](https://github.com/Azure/ARO-RP/blob/14f7393dd0c1a12bd85d79d8293bc6706701d30b/cmd/aro/update_ocp_versions.go#L60-L64) to ensure the `defaultStreams` map has one and only one item.  We can't really block RP rollouts over this but we can block OCP version updates.

Reply:
Additionally, the absence of a default version in CosmosDB is something I think SREs should be alerted to so the database can be corrected.  Maybe a followup task would be to add an `OpenShiftVersions` changefeed listener in the metrics code and emit a default version metric that we can craft an alert around?

Reply:
This is something that would not be changing continuously. I believe only simple test added to our e2e suite verifying the presence would be sufficient. Running e2e is mandatory step after rollout. This would alert us soon enough making manual corrective action possible.